package command_message.msgs;
// import "gz_string.proto";
import "pose.proto";

message CommandMessage
{
	// An indicator of what kind of message is being sent
	// 0. For accomplished feedback from model
	// 1. For indicating a connection has been established
	// 2. For Position Setting (Planar motion command)
	// 3. For Joint angle Setting (Gaits table)
	// 4. For direct control command
	// 5. For Joint Initiation
	// 6. For Model Destruction

	// 10. For require an error information
	// 11. For require a joint angle information of model
	// 12. For require a position information of model

	// 20. For error information from model
	// 21. For joint angle information from model
	// 22. For position information from model
	required int32 MessageType = 1;
	// A time based gait table timer variable, unit: ms
	optional uint32 Timer = 2;
	// A string of the model name/error information/message ID
	optional string StringMessage = 6;
	// A Position that model will drive to 
	// Or a position that system acquired
	optional gazebo.msgs.Pose PositionNeedToBe = 5;
	// The gait table of the joint angles/speed/increment need to be set
	// The unit of the joint position is : rad
	// The unit of the joint speed is: read/s
	repeated double JointGaitTable = 3 [packed=true];
	// A table indicates which joint need to be set
	// Array of four values
	// Meaning of each value in message type 3:
	// 0: p; 1: s; 2: t; 3: i; 4: c; 5: d
	// Meaning of each value in message type 4:
	// 0: this joint not being controller; other: joint being controlled
	repeated int32 JointGaitTableStatus =4 [packed=true];
}